pkg/connector/client/pagination.go

```diff
@@ -2,70 +2,232 @@ package client
+// The function calculates the next offset and serializes it into a `UserPagination` JSON string.
+// Which creates the token needed by the baton-sdk to request the subsequent page of users.
+// This implementation returns an empty string when the last page is reached, signaling the end of pagination.
+func GetUserNextToken(offset, limit, total int) string {
+	logger := zap.L()
 	nextOffset := offset + limit
+
+	logger.Debug("GetUserNextToken called",
+		zap.Int("offset", offset),
+		zap.Int("limit", limit),
+		zap.Int("total", total),
+		zap.Int("nextOffset", nextOffset),
+	)
+
 	if nextOffset >= total {
+		logger.Debug("GetUserNextToken: pagination complete",
+			zap.Int("nextOffset", nextOffset),
+			zap.Int("total", total),
+		)
+		return ""
+	}
+
+	bytes, err := json.Marshal(UserPagination{Offset: nextOffset})
+	if err != nil {
+		logger.Error("GetUserNextToken: failed to marshal pagination token",
+			zap.Int("nextOffset", nextOffset),
+			zap.Error(err),
+		)
 		return ""
 	}
 
-	bytes, err := json.Marshal(
-		Pagination{
-			Offset:          nextOffset,
-			PagingRequestId: pagingRequestId,
-		},
+	nextToken := string(bytes)
+	logger.Debug("GetUserNextToken: token generated",
+		zap.String("nextToken", nextToken),
+	)
+
+	return nextToken
+}
+
+// ParseContentPaginationToken function decodes the pagination token for the content discovery API.
+// It implements the token parsing for Percipio's non-standard, stateful content pagination.
+// The function deserializes the JSON pagination token and extracts the `Offset`, `PagingRequestId`, and `FinalOffset`.
+// Which allows the connector to maintain the complex state required between calls to the content endpoint.
+// This implementation is specific to the unique requirements of the `/catalog-content` API.
+func ParseContentPaginationToken(pToken *pagination.Token) (int, string, int, error) {
+	var (
+		offset          = 0
+		pagingRequestId = ""
+		finalOffset     = 0
 	)
+
+	if pToken != nil && pToken.Token != "" {
+		var parsed ContentPagination
+		err := json.Unmarshal([]byte(pToken.Token), &parsed)
+		if err != nil {
+			return 0, "", 0, err
+		}
+		offset = parsed.Offset
+		pagingRequestId = parsed.PagingRequestId
+		finalOffset = parsed.FinalOffset
+	}
+
+	return offset, pagingRequestId, finalOffset, nil
+}
+
+// GetContentNextToken function generates the next pagination token for the content discovery API.
+// It implements the token generation for Percipio's non-standard, stateful content pagination.
+// The function calculates the next offset and serializes it along with the required `PagingRequestId` and `FinalOffset` into a JSON string.
+// Which creates the stateful token needed to request the subsequent page of content.
+// This implementation returns an empty string when the final offset is reached, signaling the end of pagination.
+func GetContentNextToken(currentOffset, limit, finalOffset int, pagingRequestId string) string {
+	logger := zap.L()
```

Comment:
we get the the logger via ctxzap.extract:

```
l := ctxzap.Extract(ctx)
```

we don't call it `logger` and we don't use L